dkg/sync/server.go

```diff
@@ -73,6 +74,13 @@ func (s *Server) AwaitAllConnected(ctx context.Context) error {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-timer.C:
+			s.mu.Lock()
+			if s.errResponse {
+				s.mu.Unlock()
```

Comment:
why unlock here and then unlock on line 82 again?

Reply:
can simply defer unlock

Reply:
line 79 unlocks the mutex if the condition is true because after that we have to return the error while line 82 unlocks if the condition is false so that other threads can lock mutex